### PR TITLE
Support parsing TokuDB storage engine's ROW_FORMAT

### PR DESCRIPTION
**Pull request by sunxiaoguang**
Support parsing TokuDB storage engine's ROW_FORMAT. Also fixed a test in terror_test.go on Go 1.12

### What problem does this PR solve? 
Support parsing TokuDB storage engine's ROW_FORMAT

### What is changed and how it works?
Change parser to support more row formats

### Check List 

Tests 
 - Unit test

### CODE REVIEW

**Comment by GregoryIan** [comment]
@kennytm @tiancaiamao PTAL

**Comment by kennytm** [comment]
@sunxiaoguang Thanks for the contribution. Is there a corresponding issue, or PR on the https://github.com/pingcap/tidb repository?

**Comment by sunxiaoguang** [comment]
> @sunxiaoguang Thanks for the contribution. Is there a corresponding issue, or PR on the https://github.com/pingcap/tidb repository?

I had issues with dm when migrating from MySQL with tables using TokuDB as storage engines. This PR adds support to TokuDB related ROW_FORMAT sql syntax.

File: parser.y [hunk 1]
```diff
@@ -414,6 +414,14 @@ import (
 	than		"THAN"
 	timeType	"TIME"
 	timestampType	"TIMESTAMP"
+	tokudbDefault "TOKUDB_DEFAULT"
```
Comment (kennytm) [inline parser.y]: Should use `tab` to separate the two columns here.

File: parser.y [hunk 2]
```diff
@@ -6770,6 +6779,38 @@ RowFormat:
 	{
 		$$ = ast.RowFormatCompact
 	}
+| "ROW_FORMAT" EqOpt "TOKUDB_DEFAULT"
```
Comment (kennytm) [inline parser.y]: (Here also)

File: parser.y (hunk 1 above)
Comment (sunxiaoguang) [reply parser.y]: Fixed

File: parser.y (hunk 2 above)
Comment (sunxiaoguang) [reply parser.y]: Fixed

**Comment by kennytm** [comment]
PTAL @tiancaiamao

**Comment by tiancaiamao** [comment]
You don't add those changes to support TokuDB storage engine's ROW_FORMAT? @sunxiaoguang 

```
mysql> create table t (id int) ENGINE = tokudb;
Query OK, 0 rows affected (0.02 sec)
```

**Comment by sunxiaoguang** [comment]
> You don't need to add those changes to support TokuDB storage engine's ROW_FORMAT? @sunxiaoguang
> 
> ```
> mysql> create table t (id int) ENGINE = tokudb;
> Query OK, 0 rows affected (0.02 sec)
> ```
the dumped script will have things like 'ROW_FORMAT = TOKUDB_DEFAULT'. And it is ROW_FORMAT part not parsable.

**Comment by sunxiaoguang** [comment]
> You don't need to add those changes to support TokuDB storage engine's ROW_FORMAT? @sunxiaoguang
> 
> ```
> mysql> create table t (id int) ENGINE = tokudb;
> Query OK, 0 rows affected (0.02 sec)
> ```

mysql> CREATE TABLE `tb704` (
    ->   `rkey` varbinary(255) NOT NULL,
    ->   `family_id` bigint(20) NOT NULL,
    ->   `qualifier` varbinary(64) NOT NULL,
    ->   `timestamp` bigint(20) NOT NULL,
    ->   `value` mediumblob NOT NULL,
    ->   `flag` varbinary(1) NOT NULL,
    ->   PRIMARY KEY (`rkey`,`family_id`,`qualifier`,`timestamp`)
    -> ) ENGINE=TokuDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin ROW_FORMAT=TOKUDB_SNAPPY;
ERROR 1105 (HY000): line 9 column 79 near "" (total length 365)

File: parser.y [hunk 3]
```diff
@@ -3035,7 +3043,8 @@ UnReservedKeyword:
 | "DYNAMIC"| "END" | "ENGINE" | "ENGINES" | "ENUM" | "ERRORS" | "ESCAPE" | "EXECUTE" | "FIELDS" | "FIRST" | "FIXED" | "FLUSH" | "FOLLOWING" | "FORMAT" | "FULL" |"GLOBAL"
 | "HASH" | "HOUR" | "LESS" | "LOCAL" | "LAST" | "NAMES" | "OFFSET" | "PASSWORD" %prec lowerThanEq | "PREPARE" | "QUICK" | "REDUNDANT"
 | "ROLE" |"ROLLBACK" | "SESSION" | "SIGNED" | "SNAPSHOT" | "START" | "STATUS" | "SUBPARTITIONS" | "SUBPARTITION" | "TABLES" | "TABLESPACE" | "TEXT" | "THAN" | "TIME" %prec lowerThanStringLitToken
-| "TIMESTAMP" %prec lowerThanStringLitToken | "TRACE" | "TRANSACTION" | "TRUNCATE" | "UNBOUNDED" | "UNKNOWN" | "VALUE" | "WARNINGS" | "YEAR" | "MODE"  | "WEEK"  | "ANY" | "SOME" | "USER" | "IDENTIFIED"
+| "TIMESTAMP" %prec lowerThanStringLitToken | "TOKUDB_DEFAULT" | "TOKUDB_FAST" | "TOKUDB_LZMA" | "TOKUDB_QUICKLZ" | "TOKUDB_SNAPPY" | "TOKUDB_SMALL" | "TOKUDB_UNCOMPRESSED" | "TOKUDB_ZLIB"
```
Comment (tiancaiamao) [inline parser.y]: I believe those are not MySQL keywords:

```
TOKUDB_DEFAULT
TOKUDB_FAST
TOKUDB_LZMA
TOKUDB_QUICKLZ
TOKUDB_SNAPPY
TOKUDB_SMALL
TOKUDB_UNCOMPRESSED
TOKUDB_ZLIB
```

https://dev.mysql.com/doc/refman/8.0/en/keywords.html#keywords-in-current-series
Would you like to move them to the `NotKeywordToken` section if so? @sunxiaoguang

**Comment by tiancaiamao** [comment]
> You don't need to add those changes to support TokuDB storage engine's ROW_FORMAT?

Sorry, I misunderstood you... @sunxiaoguang 
This PR add `ROW_FORMAT = XXX`, not `ENGINE = XXX`

File: parser.y (hunk 3 above)
Comment (sunxiaoguang) [reply parser.y]: done

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**